### PR TITLE
XAMARIN-42843 Need to prepare UG content for SaveAsImageAsync API in Diagram

### PR DESCRIPTION
**Pull request by HemalathaDuraisamy**
Hi Team,

To include  SaveAsImageAsync API in Diagram control. Kindly review the changes and let me know your feedback.

With Regards,
Hemalatha D.

### CODE REVIEW

**Review by Gayathri-github7 — COMMENTED** [review]
Hi @HemalathaDuraisamy ,

Sender also diagram control. So caste the sender and use it. Make the changes for all API

With Regards,
Gayathri R

With Regards,
Gayathri R

File: Xamarin/Diagram/Diagram.md [hunk 1]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+private void Diagram_Loaded(object sender)
+{
+    Stream stream = new MemoryStream();
+    diagram.SaveAsImageAsync(stream, ImageFormat.Png);
+}
 {% endhighlight %}
 {% endtabs %}
 
+N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in three platforms Android, iOS and UWP.
+
```
Comment (Gayathri-github7) [inline Xamarin/Diagram/Diagram.md:279]: Hi @HemalathaDuraisamy ,

Don;t mention three platforms in works and mentioned the platform directly.

With Regards,
Gayathri R

**Review by Gayathri-github7 — COMMENTED** [review]
Hi @HemalathaDuraisamy ,

Need to mention remark for SaveAsImage API only support in Android and iOS with stream.

With Regards,
Gayathri R

**Comment by HemalathaDuraisamy** [comment]
rebuild

File: Xamarin/Diagram/Diagram.md [hunk 2]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+    Stream stream = new MemoryStream();
+    diagram.SaveAsImage(stream);
+}
+{% endhighlight %}
+{% endtabs %}
+
+N>To save the diagram as [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_) stream and [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in Android and iOS platform only.
+
+### Save the diagram as Stream Asynchronous.
+To save the diagram as Stream Asynchronous using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.
```
[the remark this answers is not in the document]
Comment (HemalathaDuraisamy) [reply Xamarin/Diagram/Diagram.md]: Hi Team,

Updated the mentioned changes.

Regards,
Hemalatha D

File: Xamarin/Diagram/Diagram.md (hunk 1 above)
Comment (HemalathaDuraisamy) [reply Xamarin/Diagram/Diagram.md:279]: Hi Team,

Updated the changes.

Regards,
Hemalatha D

File: Xamarin/Diagram/Diagram.md [hunk 3]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+{% endhighlight %}
 {% highlight c# %}
 SfDiagram diagram = new SfDiagram();
-diagram.SaveAsImage();
+diagram.Loaded += Diagram_Loaded;
 this.Content = diagram;
+
+private void Diagram_Loaded(object sender)
+{
+    diagram.SaveAsImage();
```
[the remark this answers is not in the document]
Comment (HemalathaDuraisamy) [reply Xamarin/Diagram/Diagram.md]: Hi Team,

Updated the mentioned changes.

Regards,
Hemalatha D

File: Xamarin/Diagram/Diagram.md [hunk 4]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
 ## SaveAsImage
 To save the output screen shot of the diagram control in image format using the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage) method. It uses default format image.png File Name - it is generated automatically by using the Current Date and Time.Save the diagram in a Specific ImageFormat.The following code sample explains this.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:162]: To save the output screen shot of the diagram control in image format using the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage) method. It uses the default format of image.png File Name - it is generated automatically by using the Current Date and Time. Save the diagram in a Specific Image Format. The following code sample explains this.

File: Xamarin/Diagram/Diagram.md [hunk 5]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
+private void Diagram_Loaded(object sender)
+{
+    diagram.SaveAsImage();   
+}
 {% endhighlight %}
 {% endtabs %}
 
 ### Save Diagram as image format with Padding.
 To save the diagram as image format using the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_Syncfusion_SfDiagram_XForms_ImageFormat_System_Int32_) method with the padding.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:181]: To save the diagram as image format, use the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_Syncfusion_SfDiagram_XForms_ImageFormat_System_Int32_) method with the padding.

File: Xamarin/Diagram/Diagram.md [hunk 6]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
+private void Diagram_Loaded(object sender)
+{
+    diagram.SaveAsImage(ImageFormat.Png,50); 
+}
 {% endhighlight %}
 {% endtabs %}
 
 ### Save the diagram as image format with Margin.
 To save the diagram as image format using the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_Syncfusion_SfDiagram_XForms_ImageFormat_System_Int32_System_Int32_System_Int32_System_Int32_) method with the margin.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:200]: To save the diagram as image format, use the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_Syncfusion_SfDiagram_XForms_ImageFormat_System_Int32_System_Int32_System_Int32_System_Int32_) method with the margin.

File: Xamarin/Diagram/Diagram.md [hunk 7]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
+private void Diagram_Loaded(object sender)
+{
+    diagram.SaveAsImage(ImageFormat.Png,40,0,0,0); 
+}
 {% endhighlight %}
 {% endtabs %}
 
 ### Save the diagram as Stream.
 To save the diagram as Stream using the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_) method by parsing empty stream.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:219]: To save the diagram as Stream, use the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_) method by parsing the empty stream.

File: Xamarin/Diagram/Diagram.md [hunk 8]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
+private void Diagram_Loaded(object sender)
+{
+    Stream stream = new MemoryStream();
+    diagram.SaveAsImage(stream);
+}
+{% endhighlight %}
+{% endtabs %}
+
+N>To save the diagram as [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_) stream and [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in Android and iOS platform only.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:238]: N>To save the diagram as [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_) stream and the [SaveAsImage](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImage_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in Android and iOS platform only.

File: Xamarin/Diagram/Diagram.md [hunk 9]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
+private void Diagram_Loaded(object sender)
+{
+    Stream stream = new MemoryStream();
+    diagram.SaveAsImageAsync(stream, ImageFormat.Png);
+}
 {% endhighlight %}
 {% endtabs %}
 
+N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in three platforms Android, iOS and UWP.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md]: N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in three platforms Android, iOS and UWP.

File: Xamarin/Diagram/Diagram.md [hunk 10]
```diff
@@ -161,44 +161,122 @@ diagram.LoadFromJson(JsonText);
...
+
 N> The saved image path location details, UWP - The saved image is stored in the ‘ApplicationData’ folder. You can get the application path using ‘ApplicationData.Current.LocalFolder’.(Ex:  C:\Users\AppData\Local\Packages\90e15aa5-f034-42d8-9532-0f8ac8ab27cf_bkebv2yxxpdar\RoamingState). Android - The saved image is stored in the ‘Pictures’ location with an image name starts with “Diagram” and DateAndTime (i.e., "Diagram" and "Current Date and Time") in the android device. iOS - The saved image is stored in the ‘Photos’ location with the image name ‘Diagram’ in the iPhone.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:280]: The saved image path location details, UWP - The saved image is stored in the ‘ApplicationData’ folder. You can get the application path using ‘ApplicationData.Current.LocalFolder’.(Ex:  C:\Users\AppData\Local\Packages\90e15aa5-f034-42d8-9532-0f8ac8ab27cf_bkebv2yxxpdar\RoamingState). Android - The saved image is stored in the ‘Pictures’ location with an image name starts with “Diagram” and DateAndTime (i.e., "Diagram" and "Current Date and Time") in the android device. iOS - The saved image is stored in the ‘Photos’ location with the image name ‘Diagram’ in the iPhone.

File: Xamarin/Diagram/Diagram.md [hunk 11]
```diff
@@ -230,15 +230,15 @@ this.Content = diagram;
...
 
 ### Save the diagram as Stream Asynchronous.
-To save the diagram as Stream Asynchronous using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.
+To save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.And save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) with the image format.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md]: To save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.And save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) with the image format.

File: Xamarin/Diagram/Diagram.md [hunk 12]
```diff
@@ -270,12 +270,12 @@ this.Content = diagram;
...
-    diagram.SaveAsImageAsync(stream, ImageFormat.Png);
+    (sender as SfDiagram).SaveAsImageAsync(stream, ImageFormat.Png);
 }
 {% endhighlight %}
 {% endtabs %}
 
-N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in three platforms Android, iOS and UWP.
+N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in Android, iOS and UWP platforms.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:278]: N>To save the diagram as [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) stream and [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) stream with image format methods are working in Android, iOS and UWP platforms.

File: Xamarin/Diagram/Diagram.md [hunk 13]
```diff
@@ -238,7 +238,7 @@ private void Diagram_Loaded(object sender)
...
 
 ### Save the diagram as Stream Asynchronous.
-To save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.And save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) with the image format.
+To save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.And save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) with the image format.The following code sample explains this.
```
Comment (carolinedolly) [inline Xamarin/Diagram/Diagram.md:241]: To save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_) method by parsing empty stream.And save the diagram as Asynchronous Stream using the [SaveAsImageAsync](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDiagram.XForms.SfDiagram.html#Syncfusion_SfDiagram_XForms_SfDiagram_SaveAsImageAsync_System_IO_Stream_Syncfusion_SfDiagram_XForms_ImageFormat_) with the image format.The following code sample explains this.

**Review by carolinedolly — APPROVED** [review]

**Comment by HemalathaDuraisamy** [comment]
rebuild

**Comment by HemalathaDuraisamy** [comment]
rebuild

**Review by Mohanaraja — COMMENTED** [review]
Changes are fine

**Pull request merged**